src/open_inwoner/openklant/views/contactform.py

```diff
@@ -60,12 +56,45 @@ def get_context_data(self, **kwargs):
         context["has_configuration"] = config.has_form_configuration()
         return context
 
+    def get_result_message(self, success=False):
+        if success:
+            return messages.add_message(
+                self.request, messages.SUCCESS, _("Vraag verstuurd!")
+            )
+        return messages.add_message(
+            self.request,
+            messages.ERROR,
+            _("Probleem bij versturen van de vraag."),
+        )
+
     def form_valid(self, form):
         config = OpenKlantConfig.get_solo()
         if config.register_email:
-            self.register_by_email(form, config.register_email)
+            registered_by_email = self.register_by_email(form, config.register_email)
         if config.register_contact_moment:
-            self.register_by_api(form, config)
+            registered_by_api = self.register_by_api(form, config)
+
+        # email setup
+        if config.register_email and not config.register_contact_moment:
+            if registered_by_email:
+                self.get_result_message(success=True)
+            else:
+                self.get_result_message(success=False)
+
+        # api setup
+        elif config.register_contact_moment and not config.register_email:
+            if registered_by_api:
+                self.get_result_message(success=True)
+            else:
+                self.get_result_message(success=False)
+
+        # both
+        elif config.register_email and config.register_contact_moment:
+            if registered_by_email and registered_by_api:
+                self.get_result_message(success=True)
+            else:
+                self.get_result_message(success=False)
```

Comment:
Can we simplify this? Something like:

````python
success = True
if config.register_email:
    success = self.register_by_email(form, config.register_email) and success
if config.register_contact_moment:
    success = self.register_by_api(form, config) and success

self.get_result_message(success=success)
````

Reply:
Thanks for the nice and simple solution..